### ISSUE TITLE
Compilation error in test/Kernel_23 with recent Boost

### ISSUE THREAD

**Issue by lrineau** [CGAL/cgal#2530]
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.12-Ic-59/Kernel_23/TestReport_lrineau_ArchLinux-CXX14.gz

```
[ 50%] Building CXX object CMakeFiles/test_result_of.dir/test_result_of.cpp.o
/usr/bin/c++  -DCGAL_TEST_SUITE -DCGAL_USE_GMP -DCGAL_USE_MPFR -I/home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX14/test/Kernel_23/include -I/home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX14/test/Kernel_23/../../include -I/home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX14/test/Kernel_23 -I/home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX14/include -I/mnt/testsuite/include  -Wall -std=gnu++14 -frounding-math   -o CMakeFiles/test_result_of.dir/test_result_of.cpp.o -c /home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX14/test/Kernel_23/test_result_of.cpp
In file included from /home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX14/test/Kernel_23/test_result_of.cpp:25:0:
/home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX14/test/Kernel_23/include/CGAL/_Result_of_kernel.h:45:10: fatal error: boost/tr1/functional.hpp: No such file or directory
 #include <boost/tr1/functional.hpp>
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.

```

Our MSVC-2017 platforms are also impacted.

**Comment by lrineau** [comment]
Cc @afabri 

The explanation is in the [release notes of Bosot 1.65][1]:
> *Removed Libraries in 1.65.0*
TR1

[1]: http://www.boost.org/users/history/version_1_65_1.html

Now we'll have to find a replacement...

**Comment by lrineau** [comment]
Actually the test was only doing something in C++11: it was checking that the types returned by C++11 `std::result_of` were the same as those returned by `tr1::result_of` (with the implementation taken from Boost TR1).

I propose to remove the test, now.

**Comment by lrineau** [comment]
Fixed by #2646.
